src/App.tsx

```diff
@@ -1,95 +1,47 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useContext } from 'react';
 
-export const App: React.FC = () => {
-  return (
-    <div className="todoapp">
-      <header className="header">
-        <h1>todos</h1>
-
-        <form>
-          <input
-            type="text"
-            data-cy="createTodo"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
-      </header>
+import { Header } from './Components/Header';
+import { Main } from './Components/Main/Main';
+import { Footer } from './Components/Footer';
+import { Todo } from './Types/todo';
+import { Filter } from './Types/filter';
+import { TodoContext } from './Components/Context/TodoContext';
 
-      <section className="main">
-        <input
-          type="checkbox"
-          id="toggle-all"
-          className="toggle-all"
-          data-cy="toggleAll"
-        />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list" data-cy="todosList">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-view" />
-              <label htmlFor="toggle-view">asdfghj</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+export const App: React.FC = () => {
+  const { todos } = useContext(TodoContext);
 
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-completed" />
-              <label htmlFor="toggle-completed">qwertyuio</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const [filter, setFilter] = useState<Filter>(Filter.All);
 
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" id="toggle-editing" />
-              <label htmlFor="toggle-editing">zxcvbnm</label>
-              <button type="button" className="destroy" data-cy="deleteTodo" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
+  const completed = todos.filter(todo => todo.completed);
+  const active = todos.filter(todo => !todo.completed);
```

Comment:
move filtering inside switch case since there is no need to filter active todos if user will not click it

Reply:
i need active before user's click here:

  return (
    <footer className="footer">
      <span className="todo-count" data-cy="todosCounter">
        {active.length === 1
          ? `${active.length} item left`
          : `${active.length} items left`}
      </span>